lib/livebook/storage/ets.ex

```diff
@@ -73,7 +76,13 @@ defmodule Livebook.Storage.Ets do
 
   @impl GenServer
   def init(_opts) do
-    table = :ets.new(@table_name, [:named_table, :protected, :duplicate_bag])
+    # Make sure that this process does not terminate abruptly
+    # in case it is persisting to disk. terminate/2 is still a no-op.
+    Process.flag(:trap_exit, true)
+
+    # enable passing table name for testing purposes
+    table = load_or_create_table(config_file_path())
+    :persistent_term.put(__MODULE__, table)
 
     {:ok, %{table: table}}
```

Comment:
Should we always read from the persistent term to have a single source of truth, or does it have a performance penalty? @josevalim

Reply:
From the erlang documentation `persistent_term has been highly optimized for reading terms at the expense of writing and updating terms` so I believe that it is alright :)

Reply:
Plus I would also say both approaches are fine. It would be fine to keep the `table` as part of the GenServer state and access it through there.

Reply:
@Qizot your call then! :D